lib/controller.ex

```diff
@@ -56,87 +55,67 @@ defmodule Controller do
-          "I" ->
-            recipe_list = Messages.get_recipe(:all) |> Formatter.numbered_list()
-            %{context | content: recipe_list, view: :index}
-
-          "Q" ->
-            nil
-
-          :welcome ->
-            context
+  def fetch_content(%{input: "Q", view: _}) do
+    nil
+  end
 
-          _ ->
-            error = Messages.get_prompt(:unknown)
-            %{context | error: error}
-        end
+  def fetch_content(%{input: "I", view: _} = context) do
+    content = Messages.get_recipe(:all) |> Formatter.numbered_list()
+    %{context | content: content, view: :index}
+  end
 
-      :index ->
-        case input do
-          "Q" ->
-            nil
-
-          _ ->
-            if Messages.get_recipe(input) do
-              recipe =
-                Messages.get_recipe(input)
-                |> RecipeParser.read_file()
-
-              %{context | content: recipe, view: :view_recipe}
-            else
-              error = Messages.get_recipe(:not_found)
-              %{context | error: error}
-            end
-        end
+  def fetch_content(%{input: _, view: :grocery_list} = context) do
+    fetch_unknown_error(context)
+  end
 
-      :view_recipe ->
-        case input do
-          "G" ->
-            if Messages.get_recipe("1") do
-              grocery_list =
-                Messages.get_recipe("1")
-                |> RecipeParser.parse_grocery_list()
-                |> Formatter.bulleted_list()
-
-              %{context | content: grocery_list, view: :grocery_list}
-            else
-              error = Messages.get_recipe(:not_found)
-              %{context | error: error}
-            end
-
-          "I" ->
-            content = Messages.get_recipe(:all) |> Formatter.numbered_list()
-            %{context | content: content, view: :index}
-
-          "Q" ->
-            nil
-
-          _ ->
-            error = Messages.get_prompt(:unknown)
-            %{context | error: error}
-        end
+  def fetch_content(%{input: input, view: :welcome} = context) do
+    case input do
+      :welcome ->
+        context
 
-      :grocery_list ->
-        case input do
-          "I" ->
-            content = Messages.get_recipe(:all) |> Formatter.numbered_list()
-            %{context | content: content, view: :index}
+      _ ->
+        fetch_unknown_error(context)
+    end
+  end
 
-          "Q" ->
-            nil
+  def fetch_content(%{input: input, view: :index} = context) do
+    if Messages.get_recipe(input) do
+      recipe =
+        Messages.get_recipe(input)
+        |> RecipeParser.read_file()
 
+      %{context | content: recipe, view: :view_recipe, last_input: input}
+    else
+      fetch_not_found_error(context)
+    end
+  end
 
-          _ ->
-            error = Messages.get_prompt(:unknown)
-            %{context | error: error}
+  def fetch_content(%{input: input, view: :view_recipe, last_input: last_input} = context) do
+    case input do
+      "G" ->
```

Comment:
🙀 Magic string! What does `G` mean?

Reply:
Got it - need to make some related changes to how I validate and store servings, last_input, and input in this next story. Will fix in next PR!